app_backend/balance_test.py

```diff
@@ -0,0 +1,97 @@
+from balance import BalanceTracker
+
+
+def test_set_balance():
+    test_cases = [
+        ("name1", 3000, 50),
+        ("name2", 4000.90, 60),
+        ("name3", 200.00, 30)
+    ]
+
+    for user_info in test_cases:
+        Tester = BalanceTracker(user_info[0], user_info[1], user_info[2])
+        assert Tester.name  == user_info[0]
+        assert Tester.dooley_dollars == user_info[1]
+        assert Tester.meal_swipes == user_info [2]
+
+def test_update_balance():
+    test_cases = [
+        ("name1", 3000, 50),
+        ("name2", 4000, 60),
+        ("name3", 2000, -3)
+    ]
+
+    dooley_deposit = [ # test deposit
+        500, 60, 450, 44.34, -50, -235
+    ]
+
+    dooley_spending = [
+        3, 13.99, 2.50, 5.55, 300.25, 74.39, 4000, 1.99, -78.5 #test spending
+    ]
+
+    swipe_deposit = [ # test deposit
+        3, 30, 25, 10, -40, 0, 1, -3
+    ]
+    # -------- test 1: dooley spending --------
+    for test_case in test_cases:
```

Comment:
This testing is good because it does a LOT of combinations and is great for catching if anything is broken; however, if you're dynamically determining if tests should be passing or failing then are you _really_ testing? My recommendation would be that you create an object with all starter fields, and how much dooley to add, how much to subtract, same for mealswipes, and if either of those should return `False` (kind of like an "expects_err" thing). 

This way, you're explicitly stating which tests you expect to fail and which pass. You can repeatedly copy and paste your first "object" to pass the obvious test cases and fail in every possible way your program can fail.